src/constructRoutes.js

```diff
@@ -13,7 +13,14 @@ import { resolvePath } from "./matchRoute.js";
 import { find } from "./utils/find";
 
 /**
- * @typedef {InputRoutesConfigObject | HTMLElement | import('parse5').DefaultTreeDocument} RoutesConfig
+ * @typedef {InputRoutesConfigObject | Element | import('parse5').DefaultTreeDocument} RoutesConfig
+ *
+ * @typedef {{
+ *  name: string;
+ *  type: "route" | "application";
+ *  routes: Array<Route>;
+ *  path?: string;
+ * } | AppRoute } Route
```

Comment:
I don't think this `Route` type is needed. Instead, I think we can use `RouteChild` wherever `Route` is used.